x-pack/plugins/actions/server/builtin_action_types/xmatters.test.ts

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Services } from '../types';
+import { validateConfig, validateSecrets, validateParams } from '../lib';
+import { actionsConfigMock } from '../actions_config.mock';
+import { createActionTypeRegistry } from './index.test';
+import { Logger } from '../../../../../src/core/server';
+import { actionsMock } from '../mocks';
+import axios from 'axios';
+import {
+  ActionParamsType,
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  getActionType,
+  XmattersActionType,
+} from './xmatters';
+
+import * as utils from './lib/axios_utils';
```

Comment:
Hopefully we won't need these - one reason to separate the connector into this "main" module, and the one that makes the actual request `lib/post_xmatters.ts` is so that we can test them separately.  We do this in connectors like pagerduty by mocking the "lib" module completely, so we can ignore networking stuff as much as we can.  The mocking of the module is done here: https://github.com/elastic/kibana/blob/d2a5c3d5f495ec986e09e3cd0e0ec5c1c0dfd67f/x-pack/plugins/actions/server/builtin_action_types/pagerduty.test.ts#L10-L12